Modules/EMCAL/src/CalibCheck.cxx

```diff
@@ -0,0 +1,348 @@
+          avg_frac += h_det->GetBinContent(i)*h_det->GetBinCenter(i);
+        }
+        avg_frac = avg_frac/(h_det->GetEntries());
+
+        if (avg_frac < mBadThresholdFractionGoodCellsEvent) {
+          result = Quality::Bad;
+          break;
+        }
+        else if (avg_frac < mMedThresholdFractionGoodCellsEvent) result = Quality::Medium;
+      }
+    }
+  }
+
+  if (mo->getName() == "fractionGoodCellsSupermodule") {
+    auto* h = dynamic_cast<TH2*>(mo->getObject());
+    if (h->GetEntries() == 0) {
+      result = Quality::Medium;
+    }
+    else {
+      for (int j=1; j<=h->GetNbinsX(); j++){
+        auto* h_supermod = h->ProjectionY(Form("h_supermod_%d",j),j,j);
+        if (h_supermod->GetEntries() == 0)
+          continue;
+        
+        Float_t avg_frac = 0.0;
+        for (int i=1; i<=h_supermod->GetNbinsX(); i++){
+          Float_t weight = h_supermod->GetBinContent(i)*h_supermod->GetBinCenter(i);
+          avg_frac += h_supermod->GetBinContent(i)*h_supermod->GetBinCenter(i);
+        }
+        avg_frac = avg_frac/(h_supermod->GetEntries());
+
+        if (avg_frac < mBadThresholdFractionGoodCellsSupermodule) {
+          result = Quality::Bad;
+          break;
+        }
+        else if (avg_frac < mMedThresholdFractionGoodCellsSupermodule) {
+          result = Quality::Medium;
+        }
+      }
+    }
+  }
+
+  if (mo->getName() == "cellAmplitudeSupermoduleCalib_PHYS"){
+    auto* h = dynamic_cast<TH2*>(mo->getObject());
+    if (h->GetEntries() == 0) {
+      result = Quality::Medium;
+    }
+    else {
+      auto* h_allsupermod_proj = h->ProjectionX("h_allsupermod_proj");
+      h_allsupermod_proj->Scale(1.0/h_allsupermod_proj->Integral());
+      for (int i=1; i<=h->GetNbinsY(); i++){
+        auto* h_supermod = h->ProjectionX(Form("h_supermod_%d",i),i,i);
+        if (h_supermod->GetEntries() == 0) {
+          result = Quality::Medium;
+        }
+        else {
+          h_supermod->Scale(1.0/h_supermod->Integral());
+
+          Double_t chi2_SM = -1000;
+          chi2_SM = h_supermod->Chi2Test(h_allsupermod_proj,"UU NORM CHI2/NDF");
+          if (chi2_SM > mBadThresholdCellAmplitudeSupermoduleCalibPHYS) {
+            result = Quality::Bad;
+            break;
+          }
+          else if (chi2_SM > mMedThresholdCellAmplitudeSupermoduleCalibPHYS) {
+            result = Quality::Medium;
+          }
+        }
+      }
+    }
+  }
+
+  if (mo->getName() == "cellTimeSupermoduleCalib_PHYS") {
+    auto* h = dynamic_cast<TH2*>(mo->getObject());
+    if (h->GetEntries() == 0) {
+      result = Quality::Medium;
+    }
+    else {
+      auto* h_allsupermod_proj = h->ProjectionX("h_supermod_proj");
+      if (h_allsupermod_proj->GetEntries() == 0) {
+        result = Quality::Medium;
+      }
+      else {
+        Double_t mean_allsupermod = h_allsupermod_proj->GetMean();
+        if (mean_allsupermod > mBadThresholdCellTimeSupermoduleCalibPHYS) {
```

Comment:
Shouldn't we also have a lower bound here? The mean can also be shifted to lower times compared to  what we expect